nvi-commons/src/main/resources/publication_query.sparql

```diff
@@ -0,0 +1,188 @@
+PREFIX : <https://nva.sikt.no/ontology/publication#>
+# This builds a flattened and simplified version of an expanded publication.
+# We assume the JSON contains a single publication object, and we extract the relevant fields.
+# In order to handle edge-cases and to guard against missing data, we use tricks like
+# OPTIONAL, COALESCE, and SAMPLE to ensure the output is as complete as possible.
+# This means that even if a field is set as OPTIONAL in this query, it probably should never be
+# missing in production data.
+
+CONSTRUCT {
+  # Build a simplified version of the main publication object
+  ?publication a :Publication ;
+  :identifier ?identifier ;
+  :title ?title ;
+  :status ?status ;
+  :language ?language ;
+  :publicationDate ?date ;
+  :publicationType ?instanceType ;
+  :isApplicable ?isApplicable ;
+  :isInternationalCollaboration ?isInternationalCollaboration ;
+  :publicationChannel ?channel ;
+  :contributor ?personId ;
+  :topLevelOrganization ?topLevelOrganization ;
+  :modifiedDate ?modifiedDate .
+
+  # Include publication channels
+  ?channel a :PublicationChannel ;
+  :channelType ?channelType ;
+  :identifier ?channelIdentifier ;
+  :name ?channelName ;
+  :scientificValue ?scientificValue ;
+  :year ?channelYear ;
+  :onlineIssn ?onlineIssn ;
+  :printIssn ?printIssn .
+
+  # Include the publication date
+  ?date a :PublicationDate ;
+  :year ?year ;
+  :month ?month ;
+  :day ?day .
+
+  # Include the organization hierarchy
+  ?topLevelOrganization a :Organization .
+  ?organization a :Organization ;
+  :countryCode ?country ;
+  :label ?label ;
+  :partOf ?parent ;
+  :hasPart ?child .
+
+  # Include all contributors
+  ?personId a :Contributor ;
+  :affiliation ?affiliation ;
+  :role ?roleType ;
+  :verificationStatus ?verificationStatus ;
+  :name ?contributorName .
+}
+
+WHERE {
+  # Find the main publication object and basic fields
+  ?publication a :Publication .
+  ?publication :status ?status .
+  OPTIONAL { ?publication :identifier ?identifier }
+  OPTIONAL { ?publication :modifiedDate ?modifiedDate }
+
+  ?publication :entityDescription ?entityDescription .
+  OPTIONAL { ?entityDescription :mainTitle ?title }
+  OPTIONAL { ?entityDescription :language ?language }
+
+  # Find publication date
+  ?date a :PublicationDate .
+  ?date :year ?year .
+  OPTIONAL { ?date :month ?month . }
+  OPTIONAL { ?date :day ?day . }
+
+  # Find publication type
+  ?reference a :Reference .
+  ?reference :publicationInstance ?instance .
+  ?instance a ?instanceType .
+
+  # Find all publication channels
+  VALUES ?validChannelTypes { :Publisher :Series :Journal }
+  BIND (?validChannelTypes AS ?validChannelType)
```

Comment:
Is this `BIND` necessary?

Reply:
I think so? The point was to make the static values available inside sub-queries as well.